test/e2e-multicluster-test/multicluster_test.go

```diff
@@ -1336,5 +1336,58 @@ var _ = Describe("Test multicluster scenario", func() {
 				g.Expect(pods.Items[0].Name).Should(ContainSubstring("orphan"))
 			}).WithPolling(2 * time.Second).WithTimeout(20 * time.Second).Should(Succeed())
 		})
+
+		It("Test application revision gc block application gc", func() {
+			ctx := context.Background()
+			app := &v1beta1.Application{}
+			bs, err := os.ReadFile("./testdata/app/app-lite.yaml")
+			Expect(err).Should(Succeed())
+			Expect(yaml.Unmarshal(bs, app)).Should(Succeed())
+			app.SetNamespace(namespace)
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Create(ctx, app)).Should(Succeed())
+			}).WithPolling(2 * time.Second).WithTimeout(5 * time.Second).Should(Succeed())
+			appKey := client.ObjectKeyFromObject(app)
+			Eventually(func(g Gomega) {
+				_app := &v1beta1.Application{}
+				g.Expect(k8sClient.Get(ctx, appKey, _app)).Should(Succeed())
+				g.Expect(_app.Status.Phase).Should(Equal(common.ApplicationRunning))
+			}).WithPolling(2 * time.Second).WithTimeout(20 * time.Second).Should(Succeed())
+
+			By("Add finalizer to application revision")
+			Eventually(func(g Gomega) {
+				_rev := &v1beta1.ApplicationRevision{}
+				g.Expect(k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: appKey.Name + "-v1"}, _rev)).Should(Succeed())
+				_rev.SetFinalizers([]string{"mine"})
+				g.Expect(k8sClient.Update(ctx, _rev)).Should(Succeed())
+			}).WithPolling(2 * time.Second).WithTimeout(10 * time.Second).Should(Succeed())
+
+			By("Deleting")
+			_app := &v1beta1.Application{}
+			Expect(k8sClient.Get(ctx, appKey, _app)).Should(Succeed())
+			Expect(k8sClient.Delete(ctx, _app)).Should(Succeed())
+
+			By("Check application existing after rt recycled")
+			Eventually(func(g Gomega) {
```

Comment:
I think here we want to check application recycling is blocked for its apprev is not deleted. So here I think the async assertion func should be `Consistently`. It means the assertion is true for a period of time. 
If `Eventually` is used, the first time the check passed will pass the whole assertion. Image that app is not deleted because it hasn't been reconciled which can pass the case.   The PR purpose -- application is deleted only after its apprev is deleted. -- is not actually tested.

Reply:
Simply replacing `Eventually` with `Consistently` could lead to first time failure. Splitted in to one `Eventually` and one `Consistently`.